src/MongoDB.Driver.Core/Core/Authentication/ScramCache.cs

```diff
@@ -0,0 +1,144 @@
+/* Copyright 2019–present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Linq;
+using System.Security;
+using MongoDB.Driver.Core.Misc;
+using MongoDB.Shared;
+
+namespace MongoDB.Driver.Core.Authentication
+{
+    /// <summary>
+    /// A cache for Client and Server keys, to be used during authentication.
+    /// </summary>
+    internal class ScramCache
+    {
+        private ScramCacheKey _cacheKey;
+        private ScramCacheEntry _cachedEntry;
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="key"></param>
+        /// <param name="entry"></param>
+        /// <returns></returns>
+        public bool TryGet(ScramCacheKey key, out ScramCacheEntry entry)
+        {
+            if (_cacheKey.Equals(key))
+            {
+                entry = _cachedEntry;
+                return true;
+            }
+            else
+            {
+                entry = null;
+                return false;
+            }
+        }
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="key"></param>
+        /// <param name="entry"></param>
+        public void Set(ScramCacheKey key, ScramCacheEntry entry)
```

Comment:
Let's call this method `Add`.

Normally we talk about "adding" things to a cache, not "setting" them.

It's coincidental that the current cache only holds one item, and that the cache eviction policy is trivial (always eject the previous value when adding a new one).

Reply:
Done